torch/distributions/studentT.py

```diff
@@ -0,0 +1,52 @@
+from numbers import Number
+import torch
+import math
+from torch.distributions.distribution import Distribution
+from torch.distributions import Chi2
+from torch.distributions.utils import broadcast_all
+
+
+class StudentT(Distribution):
+    r"""
+    Creates a Student's t-distribution parameterized by `df`.
+
+    Example::
+
+        >>> m = StudentT(torch.Tensor([2.0]))
+        >>> m.sample()  # Student's t-distributed with degrees of freedom=2
+         0.1046
+        [torch.FloatTensor of size 1]
+
+    Args:
+        df (float or Tensor or Variable): degrees of freedom
+    """
+    has_rsample = True
+
+    def __init__(self, df):

```

Comment:
Tensorflow additionally has `loc=0`,`scale=1` parameters, and I think these are quite useful in practice where Student's t is used for "Gaussians with outliers". I think it wouldn't be much work to add those parameters, just a couple affine transforms and a `log(scale)` term in the `log_prob` and `entropy`? What do you think?

Reply:
Sure I can add it. There are a number of distributions in addition to student's T to which we can add `loc and scale`. Should I make a pass over and add them to add applicable ones?

Reply:
Only if it helps improve compatibility with Tensorflow Distributions. I'm not a fan of scipy's design choice to add a `loc` parameter to nearly every distribution. I think what we would be more useful would be an `AffineBijector` and `TransformedDistribution` (working on those sections of the design doc right now, in case you want to review).